src/plugins/expressions/common/expression_types/specs/datatable.ts

```diff
@@ -112,7 +113,7 @@ interface RenderedDatatable {
 
 export const datatable: ExpressionTypeDefinition<typeof name, Datatable, SerializedDatatable> = {
   name,
-  validate: (table) => {
+  validate: (table: Record<string, unknown>) => {
```

Comment:
The `ExpressionValueBoxed` doesn't define a key type. We cannot use that without following typecasting because it has no `columns` and `rows` properties.